quixstreams/sources/community/kinesis/consumer.py

```diff
@@ -0,0 +1,190 @@
+            shards.extend(response["Shards"])
+        return shards
+
+    def _get_shard_iterator(self, shard_id: str):
+        if sequence_number := self._checkpointer.get(shard_id):
+            additional_kwargs = {
+                "ShardIteratorType": "AFTER_SEQUENCE_NUMBER",
+                "StartingSequenceNumber": sequence_number,
+            }
+        else:
+            additional_kwargs = {
+                "ShardIteratorType": self._auto_offset_reset,
+            }
+        response: GetShardIteratorOutputTypeDef = self._client.get_shard_iterator(
+            StreamName=self._stream, ShardId=shard_id, **additional_kwargs
+        )
+        return response["ShardIterator"]
+
+    def _init_shards(self):
+        if not (shards := [shard["ShardId"] for shard in self._list_shards()]):
+            raise ValueError(f"No shards for stream {self._stream}")
+        self._shard_iterators = {
+            shard: self._get_shard_iterator(shard) for shard in shards
+        }
+
+    def _poll_and_process_shard(self, shard_id):
+        """Read records from a shard."""
+        if (
+            backoff_time := self._shard_backoff.get(shard_id)
+        ) and time.monotonic() < backoff_time:
+            return
+        try:
+            response = self._client.get_records(
+                ShardIterator=self._shard_iterators[shard_id],
+                Limit=self._max_records_per_shard,
+            )
+
+            for record in response.get("Records", []):
+                self._process_record(shard_id, record)
+
+            # Update the shard iterator for the next batch
+            self._shard_iterators[shard_id] = response["NextShardIterator"]
+            self._shard_backoff[shard_id] = 0
+
+        except ClientError as e:
+            error_code = e.response["Error"]["Code"]
+            logger.error(f"Error reading from shard {shard_id}: {error_code}")
+            if error_code == "ProvisionedThroughputExceededException":
+                self._shard_backoff[shard_id] = time.monotonic() + self._backoff_secs
+            elif error_code == "ExpiredIteratorException":
+                logger.error(f"Shard iterator expired for shard {shard_id}.")
+                raise
+            else:
+                logger.error(f"Unrecoverable error: {e}")
+                raise
+
+    def start(self):
+        self._init_client()
+        self._init_shards()
+
+    def poll_and_process_shards(self):
+        for shard in self._shard_iterators:
+            self._poll_and_process_shard(shard)
+
+    def commit(self, force: bool = False):
+        self._checkpointer.commit(force=force)
+
+    def run(self):
+        """For running _without_ using Quix Streams Source framework."""
+        try:
+            self.start()
+            while True:
+                self.poll_and_process_shards()
+                self.commit()
+        except Exception as e:
+            logger.debug(f"KinesisConsumer encountered an error: {e}")
```

Comment:
1. Is it safe to ignore all the exceptions here?
2. If yes, it's better to do `logger.debug("KinesisConsumer encountered an error", exc_info=True)` so the logger formats the exception on its own.